### PR TITLE
refactor: Use parameterized in redfish unit tests.

### LINKED ISSUE #53: Refactor tests with parameterized
Refactor tests with similar inputs using the [parameterized](https://pypi.org/project/parameterized/) package. For eg: `test_redfish.py::test_get_storage_controller_data_success` and `test_redfish.py::test_non_standard_storage_uri_name` can be combined into one test.

Refer [this commit](https://github.com/canonical/hardware-observer-operator/pull/119/commits/c5c60d4fe751da414d444b59407a9d4b298d0964) for implementation details.

### PR DESCRIPTION
**Pull request by dashmage**
Use the [parameterized](https://pypi.org/project/parameterized/) package in the `test_redfish.py` unit tests. This allows us to more easily create similar looking tests which only differ in the parameters being passed to them.

Resolves #53

### CODE REVIEW

**Review by Pjack — CHANGES_REQUESTED** [review]
Thanks! It's a good refactor. And after the refactor, I also realize we miss some test case for the fail case. Could you help to add them? thanks!
get_storage_drive_data_fail
get_storage_controller_data_fail

**Comment by dashmage** [comment]
> Thanks! It's a good refactor. And after the refactor, I also realize we miss some test case for the fail case. Could you help to add them? thanks! get_storage_drive_data_fail get_storage_controller_data_fail

Sure thing, good suggestion. Although in this PR, I'd like to contain the scope and not add any new tests, just refactor existing ones to make it more cleaner using the `parameterized` package.

I'll create a new issue and will add the tests as part of that one :)

**Review by Pjack — APPROVED** [review]

File: tests/unit/test_redfish.py [hunk 1]
```diff
@@ -518,7 +511,9 @@ def mock_get_response(uri):
                 }
             # response for GET request to /redfish/v1/Systems/<sys_id>/
             elif "Systems" in uri:
```
Comment (rgildein) [inline tests/unit/test_redfish.py:513]: I do not see how this is used.

**Review by rgildein — APPROVED** [review]
I like the usage of parametrize, and this changes are LGTM to me.

File: tests/unit/test_redfish.py (hunk 1 above)
Comment (rgildein) [reply tests/unit/test_redfish.py:513]: This clarify it thank , I assumed that was there by design. Thanks

**Pull request merged**